publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceMessages.java

```diff
@@ -1,27 +1,127 @@
 package no.unit.nva.publication.service.impl;
 
+import static java.util.Objects.isNull;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.Organization.Builder;
+import no.unit.nva.model.Publication;
+import no.unit.nva.publication.model.MessageDto;
 import no.unit.nva.publication.storage.model.Message;
-import no.unit.nva.publication.storage.model.Resource;
 import nva.commons.core.JacocoGenerated;
+import nva.commons.core.JsonSerializable;
 
-public class ResourceMessages {
+public class ResourceMessages implements JsonSerializable {
 
-    private final Resource resource;
-    private final List<Message> messages;
+    public static final String EMPTY_MESSAGE_LIST_ERROR = "Message list cannot be empty";
```

Comment:
The naming here is a bit off as this is all of the communication about a publication, where the communication is in the form of Messages. I guess it is not important, but ResourceConversation (I'm avoiding "communication" is it may be misinterpreted)